odoo/tools/populate.py

```diff
@@ -59,12 +59,26 @@ def get_field_variation_date(model: Model, field: Field, factor: int, series_ali
     setting duplicated records too far back in the past.
     """
     total_days = min((MAX_DATETIME - MIN_DATETIME).days, factor)
-    return SQL("""
-        %(field)s - (%(factor)s - %(series_alias)s) * floor(%(total_days)s/%(factor)s) * interval '1 days'
-    """, field=SQL.identifier(field.name),
+    cast_type = SQL('date') if field.type == 'date' else SQL('timestamp')
+    query = SQL("""
+        (%(field)s - (%(factor)s - %(series_alias)s) * floor(%(total_days)s/%(factor)s) * interval '1 days')::%(cast_type)s
+    """, field=SQL('value::%s', SQL(cast_type)) if field.company_dependent else SQL.identifier(field.name),
          factor=factor,
          series_alias=SQL.identifier(series_alias),
-         total_days=total_days)
+         total_days=total_days,
+         cast_type=cast_type)
+    if field.company_dependent:
+        return SQL("""
+            CASE
+                WHEN %(field)s IS NOT NULL
+                THEN (
+                    SELECT jsonb_object_agg(key, %(value)s)
+                    FROM jsonb_each_text(%(field)s)
```

Comment:
I think you don't need `CASE`, just `SELECT` and it will be NULL if the field value was NULL

Reply:
Thanks for the catch, I recall adding it because it was breaking when processing NULL entries, but it might have been in another context, because now I've retested and it's works well without. So I've taken the opportunity to cleanup also in the char variation.